### DISCUSSION TITLE
How to config babel-plugin-react-css-modules with nextjs 10 (with scss)?

### DISCUSSION THREAD

**Discussion by jaisharx** [vercel/next.js#22486] [Help]
Is there anyone out here using babel-plugin-react-css-modules with create-next-app (with scss)? Would love to know the step-by-step procedure? I was able to configure the plugin with custom .babelrc file in create-next-app, but the custom pathnames it was generating was different than that of being applied on elements.

**Comment by jaisharx** [comment]
`{
    "presets": ["next/babel"],
    "plugins": [
        [
            "react-css-modules",
            {
                "filetypes": {
                    ".scss": {
                        "syntax": "postcss-scss"
                    }
                }
            }
        ]
    ]
}
`
above is the custom babelrc file I'm using as of now.

but using this generates pathnames like:- components-header-___styles-module__header___32eSb
while when I look at styles that are being inlined in head:- .styles_header__15DYH

there is a generateScopedName option available by the plugin as well, but configuring it still does not give the exact stylenames I need.

**Reply by cWorawan** [reply]
How did you set up `babel-plugin-react-css-modules` to work with Nextjs?
I'm using Next 11 right now and is struggling to find a way to make it work. I'm using SCSS as well.

So far it generates styleName into class on the browser, but there is no actual CSS being loaded.

**Comment by jaisharx** [comment]
Hey @ChawitO. I never could make the css-module babel plugin play nice with next.js. There really isn't very deep documentation available, and neither no one seems interested in the community to find a solution for it. As of now I just use chakra-UI, and their style props is really beautiful way of writing css, and that's what I will be sticking to at least some good amount of time.

**Reply by cWorawan** [reply]
Thanks, I spend quite a few hours yesterday trying to get it working. But after sleeping on it, I decided its probably not worth the effort.

Have you tried tailwindcss? How would you compare it to chakra-UI?

**Comment by jaisharx** [comment]
Yeah I have used tailwind for some mid size react projects, and tbh I would pick it over writing vanilla CSS/SCSS any day. But it's just that a component library like chakra allows us to build upon already built components as well (which ofc any of us can build any day). But if priority is to move fast and not do a lot of work yourself, it makes so much more sense. 

With tailwind now we have headless-UI which brings this pain point of writing same components over and over, but it only handles the functionality part and is also limited to small set of components. That's just my take, so do take it with a grain of salt.

**Comment by xyy94813** [comment]
https://github.com/vercel/next.js/blob/b21e3c258e24becc3c33397c023c0c9d9af59f75/packages/next/build/webpack/config/blocks/css/loaders/getCssModuleLocalIdent.ts#L1-L53

This is probably due to next's complex generation of LocalIdent.

> The generateScopedName option value MUST match localIdentName option of css-loader to ensure both Webpack and this plugin generate matching class names. The same goes for other options impacting class names (e.g. the default length of hashes generated by Webpack, which is used if you don't specify the hash length explicitly in localIdentName hash placeholders), and also the actuals version of this plugin and css-loader (see [css-loader compatibility](https://github.com/birdofpreyru/babel-plugin-react-css-modules#css-loader-compatibility)).

`@dr.pogodin/babel-plugin-react-css-modules` support some way to resolve the problem.

```js
const { getLocalIdent } = require('@dr.pogodin/babel-plugin-react-css-modules/utils');

const cssLoaderOptions = {
  modules: {
    getLocalIdent,
    localIdentName: '[path]___[name]__[local]___[hash:base64:6]'
  }
};
```

But in `next,js` v12.1.0, i don't find anyway to override `localIdentName`
